### ISSUE TITLE
Intuition on SE Block

### ISSUE THREAD

**Issue by pGit1** [titu1994/keras-squeeze-excite-network#7]
Any intuition on why use sigmoid instead of softmax layer in SE block?

**Comment by titu1994** [comment]
Because it's an excitation mechanism, not a Softmax attention mechanism. 

Softmax ensures everything sums up to 1. This means that few of the filters will be non-sparse, whereas the remaining majority of the filters will become sparse (close to 0 values). This is redundant and waste of memory. 

Excitation multiplies it to scale by a 0-1 range for all of the filters independently. Therefore, some filters will be close to 0, but most will lie into 0.1-1.0 range of excitation as per what it learns so there is no sparsity and all filters can equally contribute to the next layer.

**Comment by pGit1** [comment]
Awesome this makes perfect sense! Thanks for the elaborate explanation.

Do you think excitation mechanisms can supplant attention mechanisms are
their uses mutually exclusive? Also I tested switching the sigmoid layer to
softmax and performance completely died.

**Comment by titu1994** [comment]
Sigmoid cannot be swapped out for this model block, as you have noticed. 

Softmax is appropriate for dot product attention, not elementwise product. I have not heard of any model which incorporated both dot product attention and elementwise gating at the same time.
